pkg/binding/controller.go

```diff
@@ -267,6 +280,23 @@ func (c *Controller) run(ctx context.Context, workers int) error {
 	return nil
 }
 
+func (c *Controller) createManagedClustersInformer(ctx context.Context) {
+	informerFactory := ocm.GetOCMInformerFactory(c.ocmClientset)
+	informer := informerFactory.Cluster().V1().ManagedClusters().Informer()
+	_, err := informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		// TODO: we can be smarter on how we handle at least the delete rather than
+		// recreate all the bindings.
+		AddFunc:    func(obj interface{}) { c.evaluateBindingPolicies(ctx) },
+		UpdateFunc: func(oldObj, newObj interface{}) { c.evaluateBindingPolicies(ctx) },
```

Comment:
don't call evaluateBindingPolicies right away.
compare oldObj labels to newObj labels. only if labels have changed continue to handling, otherwise no need to do anything (we don't want to re-evaluate everything every time the cluster sends keep alive and status of MC CR is updated)

Reply:
Done

Reply:
@nirrozenbaum @effi-ofer Why only labels?  Other properties in cluster (annotations, others) don't have any impact on the placement ?   I thought you can use other things that just label selector.

Reply:
Only the labels are used according to https://github.com/kubestellar/kubestellar/issues/1679.  @nirrozenbaum please chip in if this is no longer true.